packages/graphql-hooks/src/useClientRequest.ts

```diff
@@ -56,13 +63,13 @@ function reducer(state, action) {
 }
 
 function useDeepCompareCallback(callback, deps) {
-  const ref = React.useRef()
+  const ref = React.useRef<DependencyList>()
 
   if (!dequal(deps, ref.current)) {
     ref.current = deps
   }
 
-  return React.useCallback(callback, ref.current)
+  return React.useCallback(callback, ref.current || [])
```

Comment:
this is a suspicious change

Reply:
Pushed an alternative approach to it that doesn't fallback to creating an empty list.

Reply:
not saying it is, but we have a history of introducing seemingly innocuous changes which turn then out to be breaking. so let's try to limit then as much as possible unless we're 100% sure